modules/programs/neovim.nix

```diff
@@ -164,8 +144,13 @@ in {
       };
 
       extraPython3Packages = mkOption {
-        type = with types; either extraPython3PackageType (listOf package);
-        default = (_: [ ]);
+        # In case we get a plain list, we need to turn it into a function,
+        # as expected by the function in nixpkgs.
+        # The only way to do so is to call `const`, which will ignore its input.
+        type = with types;
```

Comment:
rather than silently doing it, I would prefer to warn the user, make it work but with a deprecation notice.

Reply:
Good point, I changed this to show a warning.